db/src/main/kotlin/edu/artic/db/AppDataManager.kt

```diff
@@ -1,38 +1,221 @@
+                        } catch (e: Throwable) {
+                            e.printStackTrace()
+                        }
+                        try {
+                            appDatabase.generalInfoDao.setGeneralInfo(result.generalInfo)
+                        } catch (e: Throwable) {
+                            e.printStackTrace()
+                        }
+                        val galleries = result.galleries
+                        if (galleries?.isNotEmpty() == true) {
+                            try {
+                                appDatabase.galleryDao.addGalleries(galleries.values.toList())
+                            } catch (e: Throwable) {
+                                e.printStackTrace()
+                            }
+                        }
+                        val objects = result.objects
+                        if (objects?.isNotEmpty() == true) {
+                            try {
+                                appDatabase.objectDao.addObjects(objects.values.toList())
+                            } catch (e: Throwable) {
+                                e.printStackTrace()
+                            }
+                        }
+                        val audioFiles = result.audioFiles
+                        if (audioFiles?.isNotEmpty() == true) {
+                            try {
+                                appDatabase.audioFileDao.addAudioFiles(audioFiles.values.toList())
+                            } catch (e: Throwable) {
+                                e.printStackTrace()
+                            }
+                        }
+
+                        val tours = result.tours
+                        if (tours?.isNotEmpty() == true) {
+                            try {
+                                appDatabase.articTourDao.addTours(tours)
+                            } catch (e: Throwable) {
+                                e.printStackTrace()
+                            }
+                        }
+
+                        val mapAnnotations = result.mapAnnotations
+                        if (mapAnnotations?.isNotEmpty() == true) {
+                            try {
+                                appDatabase.articMapAnnotationDao.addAnnotations(mapAnnotations.values.toList())
+                            } catch (e: Throwable) {
+                                e.printStackTrace()
+                            }
+                        }
+
+                        try {
+                            appDatabase.dataObjectDao.setDataObject(result.data)
+                        } catch (e: Throwable) {
+                            e.printStackTrace()
+                        }
+                    }
+                    return@flatMap appDataState.asObservable()
+                }
+    }
+
+    private fun loadPostAppData(): Observable<Int> {
+        return Observable.create { observer ->
+            val maxDownloads = 2
```

Comment:
what are we doing here? what is max downloads.

Reply:
So it's not just the blob we have to download.
We also have to download the exhibits (different then blob exhibits) and events json